hydra-node/test/Hydra/HeadLogicSpec.hs

```diff
@@ -474,6 +473,35 @@ spec =
           Error RequireFailed{} -> True
           _ -> False
 
+      it "rejects same version snapshot requests with differring decommit txs" $ do
+        let decommitTx = SimpleTx 2 (utxoRef 2) (utxoRef 4)
+            activeUTxO = utxoRefs [2]
+            snapshot =
+              Snapshot
+                { headId = testHeadId
+                , version = 0
+                , number = 1
+                , confirmed = []
+                , utxo = activeUTxO
+                , utxoToDecommit = Just $ utxoRefs [3]
+                }
+            -- NOTE: Signatures are not relevant here
+            s0 =
+              inOpenState' threeParties $
+                coordinatedHeadState
+                  { confirmedSnapshot = ConfirmedSnapshot snapshot (Crypto.aggregate [])
+                  , seenSnapshot = LastSeenSnapshot 1
+                  , localUTxO = activeUTxO
+                  , decommitTx = Just $ SimpleTx 1 (utxoRef 1) (utxoRef 3)
+                  }
+            reqSn = receiveMessageFrom bob $ ReqSn 0 2 [] (Just decommitTx)
+
+        outcome <- runHeadLogic bobEnv ledger s0 $ do
+          step reqSn
```

Comment:
Could: make this a multi-step test which does `ReqSn 0 1` and `AckSn` accordingly. That would be less "whiteboxy" and not depend so much on the internal state of `confirmedSnapshot` and `seenSnapshot`